jme3-materialeditor/src/com/jme3/gde/materialdefinition/fileStructure/ShaderNodesBlock.java

```diff
@@ -29,31 +32,95 @@ public ShaderNodesBlock(Statement sta) {
             addStatement(b);
         }
     }
-    
-    public List<ShaderNodeBlock> getShaderNodes() {        
+
+    public List<ShaderNodeBlock> getShaderNodes() {
         return getBlocks(ShaderNodeBlock.class);
     }
-    
+
     public void addShaderNode(ShaderNodeBlock shaderNodeBlock) {
-        addStatement(shaderNodeBlock);      
+        addStatement(shaderNodeBlock);
         shaderNodeBlock.addPropertyChangeListener(WeakListeners.propertyChange(this, shaderNodeBlock));
     }
-    
+
     public boolean removeShaderNode(ShaderNodeBlock shaderNodeBlock) {
         return contents.remove(shaderNodeBlock);
     }
-    
+
+    @Override
     public void propertyChange(PropertyChangeEvent evt) {
-        if (evt.getPropertyName().equals("order")) {
+        if (evt.getPropertyName().equals(ShaderNodeBlock.ADD_MAPPING) || evt.getPropertyName().equals("order")) {

```

Comment:
Adding a listen for ADD_MAPPING  here so that we don't get the bubble saying the node is undeclared when adding a connection to a node higher up in the hierachy